lib/model/app_model/bkpr_listincome.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:cln_common/cln_common.dart';
+import 'package:clnapp/model/app_model/app_utils.dart';
+
+class AppListIncome {
+  int balance;
+  List<ListIncome> incomes;
+
+  AppListIncome({required this.balance, required this.incomes});
+
+  factory AppListIncome.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = false}) {
+    int tempBalance = 0;
+    LogManager.getInstance.debug("Response : $json");
+    var incomeEvents =
+        json.withKey("income_events", snackCase: snackCase) as List;
+    List<ListIncome> outputList = [];
+    if (incomeEvents.isNotEmpty) {
+      for (var income in incomeEvents) {
+        ListIncome temp = ListIncome.fromJSON(income);
+
+        /// FIXME : Can we do tempbalance = (2 * tempbalance - debit + credit) / 2
+        tempBalance += temp.creditMsat;
+        tempBalance -= temp.debitMsat;
+        outputList.add(temp);
+      }
+    }
```

Comment:
```suggestion
      for (var income in incomeEvents) {
        ListIncome temp = ListIncome.fromJSON(income);

        /// FIXME : Can we do tempbalance = (2 * tempbalance - debit + credit) / 2
        tempBalance += temp.creditMsat;
        tempBalance -= temp.debitMsat;
        outputList.add(temp);
      }
```

there is no need to check, the loop will be not run